lib/host/wasi/inode-win.cpp

```diff
@@ -57,6 +69,445 @@ createNullTerminatedString(std::string_view View) noexcept {
   }
   return {CStr, std::move(Buffer)};
 }
+
+inline LARGE_INTEGER toLargeIntegerFromUnsigned(unsigned long long value) {
+  LARGE_INTEGER Result;
+
+  // Does the compiler natively support 64-bit integers?
+#ifdef INT64_MAX
+  Result.QuadPart = static_cast<int64_t>(value);
+#else
+  Result.high_part = value & 0xFFFFFFFF00000000;
+  Result.low_part = value & 0xFFFFFFFF;
+#endif
+  return Result;
+}
+
+inline LARGE_INTEGER toLargeIntegerFromSigned(long long value) {
+  LARGE_INTEGER Result;
+
+#ifdef INT64_MAX
+  Result.QuadPart = static_cast<int>(value);
+#else
+  Result.high_part = value & 0xFFFFFFFF00000000;
+  Result.low_part = value & 0xFFFFFFFF;
+#endif
+  return Result;
+}
+
+inline constexpr __wasi_errno_t fromWinError(DWORD winerr) {
+  __wasi_errno_t error = __WASI_ERRNO_NOSYS;
+  switch (winerr) {
+  case ERROR_ACCESS_DENIED:
+    error = __WASI_ERRNO_ACCES;
+    break;
+  case ERROR_ACCOUNT_DISABLED:
+    error = __WASI_ERRNO_ACCES;
+    break;
+  case ERROR_ACCOUNT_RESTRICTION:
+    error = __WASI_ERRNO_ACCES;
```

Comment:
If there are several cases return the same error number, do the following stuff:

```

  case ERROR_ACCESS_DENIED:
  case ERROR_ACCOUNT_DISABLED:
  case ERROR_ACCOUNT_RESTRICTION:
    error = __WASI_ERRNO_ACCES;
```